### PR TITLE
Resilient infra 1

### PR DESCRIPTION
**Pull request by tisnik**
@GeetikaBatra I'd need your help. Could you please run the new tests against your dev. instance and check the results? Thank you in advance.

### CODE REVIEW

**Comment by GeetikaBatra** [comment]
Sure @tisnik.

**Comment by tisnik** [comment]
@GeetikaBatra FYI: I've added more chaos-monkey-like tests that needs to be checked.

File: integration-tests/features/steps/oc.py [hunk 1]
```diff
@@ -83,3 +83,9 @@ def oc_service_exist(context, service_name):
     metadata = check_and_get_attribute(context.oc_result, "metadata")
     name = check_and_get_attribute(metadata, "name")
     assert name == service_name, "Returned service has wrong name {name}".format(name=name)
+
+
+@when(u'I delete all pods for the service {service_name}')
+def oc_delete_selected_pods(context, service_name):
+    """Delete selected pods."""
+    oc_delete_pods(service=service_name)
```
Comment (GeetikaBatra) [inline integration-tests/features/steps/oc.py]: oc_delete_pods(service=service_name) didn't work for me, I changed it to ` oc_delete_pods(service_name)`, but still I get this as result.
https://gist.github.com/GeetikaBatra/fbe75eb078c7f2293165f4122c8c6ef0

Comment (GeetikaBatra) [reply integration-tests/features/steps/oc.py]: https://github.com/fabric8-analytics/fabric8-analytics-common/blob/master/integration-tests/features/src/utils.py#L62 . The oc_delete_pod function doesnot take service as a formal parameter. I think we need to introduce service=None here .

**Comment by tisnik** [comment]
You are right @GeetikaBatra , the syntax needs to be improved.

**Comment by msrb** [comment]
@tisnik fixed deleting pods with `--grace-period=0` -- that will do the trick. Note it may take 2-3 minutes for the new pod to start as if it gets scheduled on a node where the docker image is not present, then it takes some time to download it. We should take it into account while waiting for results.

**Comment by msrb** [comment]
Also README needs to be updated so people know how to run these tests.

**Comment by tisnik** [comment]
Thanks a lot @msrb, I'm on it.

**Comment by tisnik** [comment]
[test]

**Comment by tisnik** [comment]
@GeetikaBatra is your change request still relevant?

**Comment by msrb** [comment]
@tisnik not sure what the status is here, but we definitely need to update README with instructions on how to actually run these tests.

**Comment by GeetikaBatra** [comment]
@msrb I am trying to run these tests, by logging into opensift using `oc login` and then in features.list commenting all the tests except `resilient_infrastructure.feature`.  I am facing the following issue.
```
@requires.openshift.console.access
    Given The OpenShift Client is installed                           # features/steps/oc.py:14 0.000sure.feature:10
    When I ask for statuses of all deployments                        # Noneures/steps/oc.py:14
['docker-compose', '-f', '/Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.yml', '-f', '/Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.integration-tests.yml', 'logs']
WARNING: Dependency conflict: an older version of the 'docker-py' package may be polluting the namespace. If you're experiencing crashes, run the following command to remedy the issue:
pip uninstall docker-py; pip uninstall docker; pip install docker
ERROR: .IOError: [Errno 2] No such file or directory: u'/Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.yml'
Exception Exception: Failed to dump server logs. Command "docker-compose -f /Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.yml -f /Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.integration-tests.yml logs --tail=50" failed:
None
Traceback (most recent call last):
  File "features/environment.py", line 585, in after_scenario
    _dump_server_logs(context, int(context.tail_logs))
  File "features/environment.py", line 161, in _dump_server_logs
    subprocess.check_call(cmd, stderr=subprocess.STDOUT)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['docker-compose', '-f', '/Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.yml', '-f', '/Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.integration-tests.yml', 'logs', '--tail=50']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/gbatra/fabric8/fabric8-analytics-common/integration-tests/venv/bin/behave", line 11, in <module>
    sys.exit(main())
  File "/Users/gbatra/fabric8/fabric8-analytics-common/integration-tests/venv/lib/python3.6/site-packages/behave/__main__.py", line 109, in main
    failed = runner.run()
  File "/Users/gbatra/fabric8/fabric8-analytics-common/integration-tests/venv/lib/python3.6/site-packages/behave/runner.py", line 672, in run
    return self.run_with_paths()
  File "/Users/gbatra/fabric8/fabric8-analytics-common/integration-tests/venv/lib/python3.6/site-packages/behave/runner.py", line 693, in run_with_paths
    return self.run_model()
  File "/Users/gbatra/fabric8/fabric8-analytics-common/integration-tests/venv/lib/python3.6/site-packages/behave/runner.py", line 483, in run_model
    failed = feature.run(self)
  File "/Users/gbatra/fabric8/fabric8-analytics-common/integration-tests/venv/lib/python3.6/site-packages/behave/model.py", line 523, in run
    failed = scenario.run(runner)
  File "/Users/gbatra/fabric8/fabric8-analytics-common/integration-tests/venv/lib/python3.6/site-packages/behave/model.py", line 919, in run
    runner.run_hook('after_scenario', runner.context, self)
  File "/Users/gbatra/fabric8/fabric8-analytics-common/integration-tests/venv/lib/python3.6/site-packages/behave/runner.py", line 405, in run_hook
    self.hooks[name](context, *args)
  File "/Users/gbatra/fabric8/fabric8-analytics-common/integration-tests/venv/lib/python3.6/site-packages/behave/log_capture.py", line 221, in f
    func(context, *args)
  File "features/environment.py", line 588, in after_scenario
    format(c=' '.join(e.cmd), o=e.output))
Exception: Failed to dump server logs. Command "docker-compose -f /Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.yml -f /Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.integration-tests.yml logs --tail=50" failed:
None
```

**Comment by tisnik** [comment]
@GeetikaBatra so you use docker-compose for running the tests? I'd need to look into that. I usually run tests directly on the 'real machine'

**Comment by tisnik** [comment]
@msrb its still WIP ;)

**Comment by msrb** [comment]
I can take care of testing ;)

**Comment by tisnik** [comment]
@msrb @GeetikaBatra README.md and tests have been updated. Please don't wait for Centos CI job to finish, it in 'deadlock' state :upside_down_face:

**Comment by GeetikaBatra** [comment]
Thanks @tisnik for the updated readme. I don't think I am running them against docker-compose. I am following the readme and still getting the following.
```
['docker-compose', '-f', '/Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.yml', '-f', '/Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.integration-tests.yml', 'kill']
['docker-compose', '-f', '/Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.yml', '-f', '/Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.integration-tests.yml', 'rm', '-fv']
Exception Exception: Failed to teardown system. Command "docker-compose -f /Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.yml -f /Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.integration-tests.yml kill" failed:
b"\x1b[33mWARNING:\x1b[0m Dependency conflict: an older version of the 'docker-py' package may be polluting the namespace. If you're experiencing crashes, run the following command to remedy the issue:\npip uninstall docker-py; pip uninstall docker; pip install docker\n.IOError: [Errno 2] No such file or directory: u'/Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.yml'\n"
Traceback (most recent call last):
  File "features/environment.py", line 599, in after_all
    _teardown_system(context)
  File "features/environment.py", line 184, in _teardown_system
    subprocess.check_output(cmd, stderr=subprocess.STDOUT)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker-compose', '-f', '/Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.yml', '-f', '/Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.integration-tests.yml', 'kill']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/gbatra/fabric8/fabric8-analytics-common/integration-tests/venv/bin/behave", line 11, in <module>
    sys.exit(main())
  File "/Users/gbatra/fabric8/fabric8-analytics-common/integration-tests/venv/lib/python3.6/site-packages/behave/__main__.py", line 109, in main
    failed = runner.run()
  File "/Users/gbatra/fabric8/fabric8-analytics-common/integration-tests/venv/lib/python3.6/site-packages/behave/runner.py", line 672, in run
    return self.run_with_paths()
  File "/Users/gbatra/fabric8/fabric8-analytics-common/integration-tests/venv/lib/python3.6/site-packages/behave/runner.py", line 693, in run_with_paths
    return self.run_model()
  File "/Users/gbatra/fabric8/fabric8-analytics-common/integration-tests/venv/lib/python3.6/site-packages/behave/runner.py", line 504, in run_model
    self.run_hook('after_all', self.context)
  File "/Users/gbatra/fabric8/fabric8-analytics-common/integration-tests/venv/lib/python3.6/site-packages/behave/runner.py", line 405, in run_hook
    self.hooks[name](context, *args)
  File "/Users/gbatra/fabric8/fabric8-analytics-common/integration-tests/venv/lib/python3.6/site-packages/behave/log_capture.py", line 221, in f
    func(context, *args)
  File "features/environment.py", line 602, in after_all
    format(c=' '.join(e.cmd), o=e.output))
Exception: Failed to teardown system. Command "docker-compose -f /Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.yml -f /Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.integration-tests.yml kill" failed:
b"\x1b[33mWARNING:\x1b[0m Dependency conflict: an older version of the 'docker-py' package may be polluting the namespace. If you're experiencing crashes, run the following command to remedy the issue:\npip uninstall docker-py; pip uninstall docker; pip install docker\n.IOError: [Errno 2] No such file or directory: u'/Users/gbatra/fabric8/fabric8-analytics-common/docker-compose.yml'\n"
```

**Comment by msrb** [comment]
Ehm, what did I do here? I was trying to rebase it and resolve conflict in README.md :/

Let me fix this.

**Comment by msrb** [comment]
Still not sure what happened. I had to cherry-pick Pavel's last commits on top of my master-rebased branch.

**Comment by msrb** [comment]
I had to fix one small issue. LGTM :wink:

**Pull request merged**